src/cli/commands/bin.js

```diff
@@ -16,6 +16,12 @@ export function setFlags(commander: Object) {
 
 export function run(config: Config, reporter: Reporter, flags: Object, args: Array<string>): Promise<void> {
   const binFolder = path.join(config.cwd, config.registries[RegistryYarn.registry].folder, '.bin');
-  reporter.log(binFolder, {force: true});
+  if (args.length === 0) {
+    reporter.log(binFolder, {force: true});
+  } else {
+    for (const arg of args) {
```

Comment:
I think we should limit the argument length to 1 to keep the API surface smaller and open to future additions.

I also think we should ensure the binary exists before returning its path. Just calling `fs.realpath` on the value before returning it would probably be better for this and to ensure consistent output.